### PR TITLE
Naked GL removal work

### PR DESCRIPTION
**Pull request by jherico**
This PR does quite a bit:

* Removes gpu/GPUConfig.h from nearly every header.  
** OpenGL headers should no longer be included anywhere except where absolutely necessary (i.e. the GL backend code, or display plugin code), and almost never in headers.
** Some classes have switched from using members like QOpenGLContext to using pointers to same and forward declarations, so that we don't inadvertently get the GL headers from QT. 
* Removes the system GL headers from InterfaceConfig.h.  
* Removes some defunct classes like ProgramObject
* Converts much of the Application.cpp naked GL to GPU.  
* Comments out the shadow related naked GL code in Application.cpp
* Adds lighting functionality to the GPU library (probably needs polishing by Sam)
* Adds framebuffer blitting functionality to the GPU library (also probably needs polishing by Sam)
* Removes the old starfield code and replaces it with something that isn't quite as performant, and isn't quite as pretty to act as a placeholder for the final solution

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 005620856526bcf0f5a7a552a8f23795e7a0f8fc available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 35f7307f30c72c7c9249d017c6dc13828ec35176 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5b494c23fd4a59217eea123cb976265ea62bfde4 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0da8b27fcf67006d8be1d188d0acc6a5e500067f available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f13152b2b4d5142fc2572d7de0cb298dc83bbe6c available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by samcake** [comment]
Amazing!!!

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -3131,42 +3106,26 @@ void Application::updateShadowMap(RenderArgs* renderArgs) {
...
+    vec3( 0.6f, 0.525f, 0.525f), // diffuse
+    vec4(0.08f, 0.08f, 0.08f, 1.0f), // specular
+    vec4( 0, 0, 0, 0 ), // position
+    96 // shininess
+};
 
+void Application::setupWorldLight(RenderArgs* renderArgs) {
+    gpu::Batch batch;
+    defaultLight._position = vec4(getSunDirection(), 0);
+    batch.setLight(0, defaultLight);
```
Comment (samcake) [inline interface/src/Application.cpp]: We can actually completely remove all that  light setup

**Comment by hifi-gustavo** [comment]
Build console output for commit: 81543d918554ba26b6823b1ca8362ddbb426f3a7 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4c3cfe1e82f2d8c79ec8f86a361542dd99581837 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5159452335487f8d2150df416af32acc71f04e4c available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d4e637c5b8cedc0cadcde4fbf0c65b942a0c3464 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: de92e66635ecbb20687ac46a36cf3eb889e2c2e0 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cd677ccb18049dbb3fef2195b5633b5045dfb891 available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 23b058e4eb7d9aac31f587bcd1d2bfce6fa4528f available at: https://gist.github.com/9f68f4972a5bf9fafd7d<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5366.dmg

**Pull request merged**